core/src/main/scala/akka/persistence/dynamodb/internal/QueryDao.scala

```diff
@@ -370,17 +388,71 @@ import software.amazon.awssdk.services.dynamodb.model.QueryResponse
     client
       .query(req)
       .asScala
-      .map { response =>
+      .flatMap { response =>
         val items = response.items()
         if (items.isEmpty)
-          None
-        else
-          Some(createSerializedJournalItem(items.get(0), includePayload))
+          Future.successful(None)
+        else {
+          Future.successful(Some(createSerializedJournalItem(items.get(0), includePayload)))
+        }
       }
       .recoverWith { case c: CompletionException =>
         Future.failed(c.getCause)
       }(ExecutionContext.parasitic)
-
   }
 
+  private def followBreadcrumb(evt: SerializedJournalItem): Future[SerializedJournalItem] =
+    if (s3Fallback.nonEmpty && evt.serManifest == S3FallbackSerializer.BreadcrumbManifest) {
+      deserializeBreadcrumb(evt).flatMap { bucket =>
```

Comment:
we could use the serializer manifest to store the bucket instead of storing it in payload